pal/src/sslClient_arduino.cpp

```diff
@@ -31,6 +32,10 @@ uint8_t sslClient_connected(void)
 
 int sslClient_connect(uint32_t ipAddress, uint16_t port)
 {
+#ifdef ARDUINO_ARCH_ESP8266
+    BearSSL::X509List cert(certificates);
```

Comment:
I think this could cause an issue. `sslClient` is defined globally but the cert here is defined and given to `sslClient` as a stack variable. If anything outside of this function used cert it would crash. I'd suggest moving this line to a static declaration under the `sslClient` with any necessary #defines.

Reply:
[Link here](https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/bearssl-client-secure-class.html#object-lifetimes) which describes this kind of thing

Reply:
Moved up.